### PR TITLE
TACKLE-635: Update stakeholder/stakeholder groups to use react query

### PR DESCRIPTION
**Pull request by ibolton336**
https://issues.redhat.com/browse/TACKLE-635

### CODE REVIEW

File: pkg/client/src/app/queries/stakeholdergoups.ts [hunk 1]
```diff
@@ -0,0 +1,61 @@
+import { useMutation, useQuery, useQueryClient } from "react-query";
+import { Stakeholder, StakeholderGroup } from "@app/api/models";
```
Comment (gildub) [inline pkg/client/src/app/queries/stakeholdergoups.ts:2]: Please remove unneeded `Stakeholder`

File: pkg/client/src/app/queries/stakeholdergoups.ts [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import {
+  deleteStakeholder,
```
Comment (gildub) [inline pkg/client/src/app/queries/stakeholdergoups.ts:4]: `deleteStakeholder` is not used

File: pkg/client/src/app/queries/stakeholdergoups.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    stakeholderGroups: data || [],
+    isFetching: isLoading,
+    fetchError: error,
+    refetch,
+  };
+};
+
+export const useDeleteStakeholderGroupMutation = (
+  onSuccess: (res: any) => void,
+  onError: (err: AxiosError) => void
```
Comment (gildub) [inline pkg/client/src/app/queries/stakeholdergoups.ts:42]: No need to propagate AxiosError this far. The `deleteStakeholderGroup` could return a Promise<AxiosResponse<StakeholderGroup>> which would propagate a standard Error.
Check out task queries for examples.

File: pkg/client/src/app/pages/controls/stakeholders/stakeholders.tsx [hunk 4]
```diff
@@ -1,5 +1,5 @@
 import React, { useCallback, useEffect, useState } from "react";
```
Comment (gildub) [inline pkg/client/src/app/pages/controls/stakeholders/stakeholders.tsx:1]: `useCallback`, `useEffect` are not used

File: pkg/client/src/app/pages/controls/stakeholders/stakeholders.tsx [hunk 5]
```diff
@@ -35,10 +35,9 @@ import {
   ConditionalRender,
   NoDataEmptyState,
 } from "@app/shared/components";
-import { useFetchStakeholders, useDelete } from "@app/shared/hooks";
+import { useDelete } from "@app/shared/hooks";
```
Comment (gildub) [inline pkg/client/src/app/pages/controls/stakeholders/stakeholders.tsx:38]: `useDelete` is not used

File: pkg/client/src/app/pages/controls/stakeholder-groups/stakeholder-groups.tsx [hunk 6]
```diff
@@ -59,6 +50,12 @@ import {
 import { useFilterState } from "@app/shared/hooks/useFilterState";
 import { useSortState } from "@app/shared/hooks/useSortState";
 import { controlsWriteScopes, RBAC, RBAC_TYPE } from "@app/rbac";
+import {
+  useDeleteStakeholderGroupMutation,
+  useFetchStakeholderGroups,
+} from "@app/queries/stakeholdergoups";
+import { deleteStakeholderGroup } from "@app/api/rest";
```
Comment (gildub) [inline pkg/client/src/app/pages/controls/stakeholder-groups/stakeholder-groups.tsx:57]: `deleteStakeholderGroup` is not used

File: pkg/client/src/app/pages/controls/stakeholder-groups/stakeholder-groups.tsx [hunk 7]
```diff
@@ -59,6 +50,12 @@ import {
...
+import { useDeleteStakeholderMutation } from "@app/queries/stakeholders";
```
Comment (gildub) [inline pkg/client/src/app/pages/controls/stakeholder-groups/stakeholder-groups.tsx:58]: `useDeleteStakeholderMutation`  is not used

File: pkg/client/src/app/queries/stakeholdergoups.ts (hunk 3 above)
Comment (ibolton336) [reply pkg/client/src/app/queries/stakeholdergoups.ts:42]: If we know the error returned is an Axios Error with the deleteStakeholder call being an axios call, what would be the purpose of a generic error message being returned?

**Pull request merged**